docs/kb/kb0004-akv2aks.md

```diff
@@ -4,257 +4,156 @@ title: Sync certificates and secrets from Azure Key Vault to Azure Kubetnetes Se
-
-kubectl exec busybox-secrets-store-inline -- cat /mnt/secrets-store/foo
-# Expected output:
-# bar
-
-kubectl exec busybox-secrets-store-inline -- cat /mnt/secrets-store/cert-demo
-# Expected output:
------BEGIN PRIVATE KEY-----
-MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQDVYhtyud6rbRJT
-...
-3fic6VM3cQR9FJxBxAq4vro=
------END PRIVATE KEY-----
------BEGIN CERTIFICATE-----
-MIIDQjCCAiqgAwIBAgIQSRZYP7ncTSGCw6IEOxTIhjANBgkqhkiG9w0BAQsFADAe
-...
-5STNJyO/kEBkBMjlzZKlDkhuf4Tr1g==
------END CERTIFICATE-----
-```
-
-```powershell
-kubectl get secrets
-# Expected output:
-# NAME                                    TYPE                                  DATA   AGE
-# cert-demo                               kubernetes.io/tls                     2      9h
-# foo                                     Opaque                                1      9h
-```
+    In your values file, we'll need to add the following block of code into the global values:
+
+    ```yaml
+    global:
+      keyvault:
+        enabled: true # When enabled, it will deploy the secret store manifest to Kubernetes
+        #frontendCrt: cluedin-sample-pfx # This must match the certificate name on the AKV end. When mounted, it will appear as `cluedin-frontend-crt`
+        userAssignedIdentityID: $kvManagedIdentity # This is the guid for the kv managed identity
+        keyvaultName: $kv.name # This is your key vault name
+        tenantId: $tenantId # This is the guid of your tenant
+        secretRanges:
+        - secretName: cluedin-login-details # This is how the secret will appear within Kubernetes Secrets once synchronised
+          secretType: Opaque # For most secrets, leave as Opaque
+          secretKeys:
+            password: cluedin-login-password # The left side (Key) is the name in the Kubernetes secret object. The right side (Value) is the AKV reference which will grab its value.
+            username: cluedin-login-username
+
+    infrastructure:
+      cert-manager:
+        #enabled: false # Only set to false if using an uploaded frontend certificate.
+    ```
+    **NOTE**: If you are using the certificate upload as part of your setup, you **must** disable cert-manager by setting `enabled: false`. You also must set `frontendCrt` to a value. It will be mounted as `cluedin-frontend-crt`. This secret name cannot change.
+
+1. With all the secrets and certificates now done, the last step is to update and `secretRef`'s in your `Values.yaml` or chart to reference the new synchronised secrets.
+
+# Technical Notes
+This section will explain some of the more technical bits.
+
+- The secret csi driver is deployed at the Kubernetes level. This deploys some additional pods to each node to facilitate this. However, if your max pods limit is the default `30`, you may run into an issue after deploying as the CluedIn application is on the borderline of this value.  
+
+    Please ensure this is checked before proceeding as it may cause the cluster to not function correctly. 
```

Comment:
It is not really clear what should be checked. Maybe "Please ensure that you have checked your max pods limit before ..."?